### PR TITLE
[Task 1863278] use az config to replace AzureConfigurations.getInstance()

### PR DESCRIPTION
**Pull request by andxu**
# What's the problem?
Currently there are two kinds of config: az config and AzureConfigurations, we need to choose one to avoid confusion

# How to solve this problem?
1. Replace the read scenarios of AzureConfigurations.getInstance() to Azure.az().config().getxxx
2. Refactor AzureConfiguration to have two methods: saveAzConfig and loadToAzConfig
3. In AzureInitializer call loadToAzConfig to load the persistent data into az config and save the az config when user clicks ok on intellij settings
4. This PR also fix a `reset` doesn't work issue in intellij settings

[AB#1863278](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1863278)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/AzurePanel.java [hunk 1]
```diff
@@ -205,10 +228,38 @@ public String getHelpTopic() {
...
+
+        return !Objects.equals(data.getTelemetryEnabled(), data.getTelemetryEnabled());
     }
 
     @Override
     public void reset() {
+        setData(originalData);
+    }
+
+    private void persistentData(AzureConfiguration data) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/AzurePanel.java:257]: persist

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/AzurePanel.java [hunk 2]
```diff
@@ -205,10 +228,38 @@ public String getHelpTopic() {
...
+        final AzureConfigurations.AzureConfigurationData config = AzureConfigurations.getInstance().getState();
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/AzurePanel.java:258]: private AzureConfigurationData

Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/ui/AzurePanel.java:258]: AzureConfigurations -> in charge of persist

**Comment by andxu** [comment]
Replaced in 5660

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/settings/AzureConfigurations.java [hunk 3]
```diff
@@ -69,6 +72,36 @@ public String pluginVersion() {
...
+        final AzureConfigurations.AzureConfigurationData state = getState();
+        if (StringUtils.isNotBlank(state.installationId())) {
+            config.setTelemetryEnabled(state.allowTelemetry());
+            config.setMachineId(state.installationId());
+            config.setCloud(state.environment());
+
+            if (StringUtils.isNotBlank(state.functionCoreToolsPath()) && new File(state.functionCoreToolsPath()).exists()) {
+                config.setFunctionCoreToolsPath(state.functionCoreToolsPath());
+            }
+            config.setDatabasePasswordSaveType(state.passwordSaveType());
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/settings/AzureConfigurations.java:99]: function core tools path and password save type should out of `if (StringUtils.isNotBlank(state.installationId())) {`. right ?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AppInsightsConfigurationImpl.java [hunk 4]
```diff
@@ -36,17 +36,17 @@ public String pluginVersion() {
 
     @Override
     public String installationId() {
-        return AzureConfigurations.getInstance().getState().installationId();
+        return Azure.az().config().getMachineId();
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AppInsightsConfigurationImpl.java:39]: renamed ? how about to rename all `installationId` to `machineId`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureInitializer.java [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        if (!AzureConfigurations.getInstance().loadToAzConfig()) {
+            initializeConfigFromLegacySettings();
+        }
+        initializeDefaultValues();
+        initializeTelemetry();
+
+        CommonSettings.setUserAgent(config.getUserAgent());
+        if (StringUtils.isNotBlank(config.getCloud())) {
+            Azure.az(AzureCloud.class).setByName(config.getCloud());
+        }
```
Comment (shenqianjin) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureInitializer.java:61]: they have different levels (low) with `initializeDefaultValues`,  `initializeTelemetry()`.

Comment (shenqianjin) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/AzureInitializer.java:61]: and there is a `config.setUserAgent(userAgent)` in `initializeDefaultValues`

**Review by shenqianjin — DISMISSED** [review]

**Comment by shenqianjin** [comment]
LGTM

**Comment by andxu** [comment]
#5691

**Review by shenqianjin — APPROVED** [review]
LGTM

**Pull request merged**